polkadot/node/core/pvf/execute-worker/src/lib.rs

```diff
@@ -182,44 +149,93 @@ pub fn worker_entrypoint(
 						continue
 					},
 				};
-
-				// SAFETY: new process is spawned within a single threaded process. This invariant
-				// is enforced by tests.
-				let response = match unsafe { nix::unistd::fork() } {
-					Err(errno) => internal_error_from_errno("fork", errno),
-					Ok(ForkResult::Child) => {
-						// Dropping the stream closes the underlying socket. We want to make sure
-						// that the sandboxed child can't get any kind of information from the
-						// outside world. The only IPC it should be able to do is sending its
-						// response over the pipe.
-						drop(stream);
-						// Drop the read end so we don't have too many FDs open.
-						drop(pipe_reader);
-
-						handle_child_process(
-							pipe_writer,
-							compiled_artifact_blob,
-							executor_params,
-							params,
-							execution_timeout,
-						)
-					},
-					Ok(ForkResult::Parent { child }) => {
-						// the read end will wait until all write ends have been closed,
-						// this drop is necessary to avoid deadlock
-						drop(pipe_writer);
-
-						handle_parent_process(
-							pipe_reader,
-							child,
-							worker_pid,
-							usage_before,
-							execution_timeout,
-						)?
-					},
+				let stream_fd = stream.as_raw_fd();
+
+				let compiled_artifact_blob = Arc::new(compiled_artifact_blob);
+				let arc_executor_params: Arc<ExecutorParams> = Arc::new(executor_params.clone());
+				let params = Arc::new(params);
+				let execute_worker_stack_size = get_max_stack_size(&executor_params, 1);
+
+				cfg_if::cfg_if! {
+					if #[cfg(target_os = "linux")] {
+						use polkadot_node_core_pvf_common::worker::security;
+
+						// Number of threads here is 3:
+						// 1 - Main thread
+						// 2 - Cpu monitor thread
+						// 3 - Execute thread
+						let stack_size = get_max_stack_size(&executor_params, 3);
+
+						// SAFETY: new process is spawned within a single threaded process. This invariant
+						// is enforced by tests. Stack size being specified to ensure child doesn't overflow
+						let result = match unsafe {
+							security::clone::clone_on_worker(
+								worker_info,
+								Box::new(|| {
+									handle_child_process(
+										pipe_writer,
+										pipe_reader,
+										stream_fd,
+										Arc::clone(&compiled_artifact_blob),
+										Arc::clone(&arc_executor_params),
+										Arc::clone(&params),
+										execution_timeout,
+										execute_worker_stack_size,
+									)
+								}),
+								stack_size as usize,
```

Comment:
I think it's already a `usize`:

```suggestion
								stack_size,
```